diff --git a/deps/StructuredOptimization/Project.toml b/deps/StructuredOptimization/Project.toml
index fcbc51f..dd89574 100644
--- a/deps/StructuredOptimization/Project.toml
+++ b/deps/StructuredOptimization/Project.toml
@@ -2,9 +2,6 @@ name = "StructuredOptimization"
 uuid = "46cd3e9d-64ff-517d-a929-236bc1a1fc9d"
 version = "0.5.0"
 
-[workspace]
-projects = ["test", "docs"]
-
 [deps]
 AbstractOperators = "d9c5613a-d543-52d8-9afd-8f241a8c3f1c"
 Combinatorics = "861a8166-3701-5b0c-9a16-15d98fcdc6aa"
@@ -18,6 +15,13 @@ ProximalCore = "dc4f5ac2-75d1-4f31-931e-60435d74994b"
 ProximalOperators = "a725b495-10eb-56fe-b38b-717eba820537"
 RecursiveArrayTools = "731186ca-8d62-57ce-b412-fbd966d074cd"
 
+[sources]
+AbstractOperators = {path = "../AbstractOperators"}
+DSPOperators = {path = "../AbstractOperators/DSPOperators"}
+FFTWOperators = {path = "../AbstractOperators/FFTWOperators"}
+ProximalAlgorithms = {path = "../ProximalAlgorithms"}
+ProximalOperators = {path = "../ProximalOperators"}
+
 [compat]
 AbstractOperators = "0.4"
 Combinatorics = "1.0.2"
diff --git a/deps/StructuredOptimization/src/StructuredOptimization.jl b/deps/StructuredOptimization/src/StructuredOptimization.jl
index 0ea93a1..10145db 100644
--- a/deps/StructuredOptimization/src/StructuredOptimization.jl
+++ b/deps/StructuredOptimization/src/StructuredOptimization.jl
@@ -3,11 +3,11 @@ module StructuredOptimization
 using LinearAlgebra
 using RecursiveArrayTools
 using ProximalCore
-using AbstractOperators, DSPOperators, FFTWOperators
-using ProximalOperators
-using ProximalAlgorithms
+using ..AbstractOperators, ..DSPOperators, ..FFTWOperators
+using ..ProximalOperators
+using ..ProximalAlgorithms
 using Combinatorics: permutations, powerset
-using ProximalAlgorithms: IterativeAlgorithm, override_parameters
+using ..ProximalAlgorithms: IterativeAlgorithm, override_parameters
 
 abstract type AbstractExpression end
 
diff --git a/deps/StructuredOptimization/src/calculus/precomposeNonlinear.jl b/deps/StructuredOptimization/src/calculus/precomposeNonlinear.jl
index d0012c2..ec135f1 100644
--- a/deps/StructuredOptimization/src/calculus/precomposeNonlinear.jl
+++ b/deps/StructuredOptimization/src/calculus/precomposeNonlinear.jl
@@ -1,4 +1,4 @@
-import ProximalOperators: gradient!, gradient, preallocate # this can be removed when moved to Prox
+import ..ProximalOperators: gradient!, gradient, preallocate # this can be removed when moved to Prox
 
 export PrecomposeNonlinear
 
diff --git a/deps/StructuredOptimization/src/syntax/expressions/utils.jl b/deps/StructuredOptimization/src/syntax/expressions/utils.jl
index f4f1a34..14d33b3 100644
--- a/deps/StructuredOptimization/src/syntax/expressions/utils.jl
+++ b/deps/StructuredOptimization/src/syntax/expressions/utils.jl
@@ -1,7 +1,7 @@
 # utils
 export variables, operator, affine
 import Base: convert
-import AbstractOperators: displacement
+import ..AbstractOperators: displacement
 
 convert(::Type{Expression}, x::Variable) =
     Expression((x,), Eye(~x))
diff --git a/deps/StructuredOptimization/src/syntax/terms/term.jl b/deps/StructuredOptimization/src/syntax/terms/term.jl
index bab7a6e..e2eafce 100644
--- a/deps/StructuredOptimization/src/syntax/terms/term.jl
+++ b/deps/StructuredOptimization/src/syntax/terms/term.jl
@@ -185,7 +185,7 @@ is_op_f = [
 
 for f in is_op_f
     @eval begin
-        import AbstractOperators: $f
+        import ..AbstractOperators: $f
         $f(t::Term) = $f(operator(t))
         $f(t::TermSet) = all($f.(t))
     end
